### PR TITLE
Add CLI switch to skip plugin logo reading

### PR DESCRIPTION
**Pull request by novotnyr**
`structure`:
- Add CLI switch `-no-logos` to skip reading plugin logos from ZIPs, JARs and directories
- Do not search for additional logos when primary logo is not found
Tests:
Add customization of plugin creation in tests, especially when creating via customized factory methods of `IdePluginManager`

Terminology: plugin icon and plugin logo are used interchangeably. The official terminology is [plugin logo](https://plugins.jetbrains.com/docs/intellij/plugin-icon-file.html).

### CODE REVIEW

File: intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/jar/PluginJar.kt [hunk 1]
```diff
@@ -38,15 +42,24 @@ class PluginJar(private val jarPath: Path, private val jarFileSystemProvider: Ja
   }
 
   fun getIcons(): List<PluginIcon> {
-    return IconTheme.values().mapNotNull { theme ->
-      val iconEntryName = "$META_INF/${getIconFileName(theme)}"
+    val (defaultThemes, additionalThemes) = IconTheme.values().partition { it == IconTheme.DEFAULT }
```
Comment (YannCebron) [inline intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/jar/PluginJar.kt:45]: IMHO we can code this explicitly against known enum values

File: intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/jar/PluginJar.kt [hunk 2]
```diff
@@ -38,15 +42,24 @@ class PluginJar(private val jarPath: Path, private val jarFileSystemProvider: Ja
...
+    return listOf(defaultIcon) + additionalThemes.mapNotNull { getIcon(it) }
+  }
 
-      val iconPath = jarFileSystem.getPath(META_INF, getIconFileName(theme))
-      if (iconPath.exists()) {
-        PluginIcon(theme, iconPath.readBytes(), iconEntryName)
-      } else {
-        null
-      }
+  private fun getIcon(theme: IconTheme): PluginIcon? {
```
Comment (YannCebron) [inline intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/jar/PluginJar.kt:55]: findPluginIcon

File: intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/plugin/IdePluginManager.kt [hunk 3]
```diff
@@ -86,12 +89,13 @@ class IdePluginManager private constructor(
...
-    resourceResolver: ResourceResolver,
-    parentPlugin: PluginCreator?,
-    problemResolver: PluginCreationResultResolver
+     pluginDirectory: Path,
+     descriptorPath: String,
+     validateDescriptor: Boolean,
+     resourceResolver: ResourceResolver,
+     parentPlugin: PluginCreator?,
+     problemResolver: PluginCreationResultResolver,
+     parsingConfiguration: PluginParsingConfiguration
```
Comment (YannCebron) [inline intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/plugin/IdePluginManager.kt:98]: do we need to pass through PParsConfig everywhere here? it's a property of IdePluginManager

File: intellij-plugin-structure/tests/src/test/kotlin/com/jetbrains/plugin/structure/mocks/MockPluginsTest.kt [hunk 4]
```diff
@@ -557,6 +554,36 @@ class MockPluginsTest(fileSystemType: FileSystemType) : BasePluginManagerTest<Id
     assertEquals(lang, plugin.extensions.values.first().first().attributes.first().value)
   }
 
+  @Test
+  fun `icons are loaded`() {
+    val plugin = buildPluginSuccess(expectedWarnings) {
+      buildZipFile(temporaryFolder.newFile("plugin.jar")) {
+        dir("META-INF", metaInfDir)
```
Comment (YannCebron) [inline intellij-plugin-structure/tests/src/test/kotlin/com/jetbrains/plugin/structure/mocks/MockPluginsTest.kt:561]: what is all these `dir()` things for?

**Comment by novotnyr** [comment]
Abandoned in favour of #1044.

There are no visible performance improvements when using this CLI switch.

**Pull request closed without merging**